scripts/core/editor3/components/handlePastedText.js

```diff
@@ -101,3 +108,54 @@ const atomicBlock = (data, entity) => new ContentBlock({
     characterList: List([CharacterMetadata.create({entity})]),
     data: data
 });
+
+/**
+ * @ngdoc method
+ * @name allowEditSuggestion
+ * @param {string} operation - one of insert, backspace or delete
+ * @returns {Boolean} True if the current text don't contains a noneditable suggestion.
+ * @description Check if the current text don't contain a noneditable suggestion.
+ */
+export function allowEditSuggestion(action) {
+    const {suggestingMode, editorState} = this.props;
+    const content = editorState.getCurrentContent();
+    const selection = editorState.getSelection();
+    const types = ['DELETE_SUGGESTION', 'ADD_SUGGESTION'];
+
+    if (selection.getStartOffset() !== selection.getEndOffset()) {
```

Comment:
You probably want to use `SelectionState.isCollapsed()` which also takes into account that a selection doesn't span multiple blocks. Do we support suggestions spanning multiple blocks by the way?

Reply:
Looks like there isn't support for multiple blocks suggestions. It will be fixed in other PR,  I created a task for this (SDNTB-472).

Reply:
You can wait until I finish with multiple highlights, and if you make suggestions use that, you'd get multi-block suggestions by default.

Reply:
OK